geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyAsyncEventQueueCommand.java

```diff
@@ -0,0 +1,86 @@
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.management.cli.ConverterHint;
+import org.apache.geode.management.cli.Result;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
+import org.apache.geode.management.internal.cli.functions.DestroyAsyncEventQueueFunction;
+import org.apache.geode.management.internal.cli.functions.DestroyAsyncEventQueueFunctionArgs;
+import org.apache.geode.management.internal.cli.i18n.CliStrings;
+import org.apache.geode.management.internal.cli.result.ResultBuilder;
+import org.apache.geode.management.internal.cli.result.TabularResultData;
+import org.apache.geode.management.internal.configuration.domain.XmlEntity;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+public class DestroyAsyncEventQueueCommand implements GfshCommand {
+  public static final String DESTROY_ASYNC_EVENT_QUEUE = "destroy async-event-queue";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__HELP = "destroy an Async Event Queue";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__ID = "id";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__ID__HELP =
+      "ID of the queue to be destroyed.";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__GROUP__HELP =
+      "Group(s) of members on which to destroy the async event queue.";
+
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__AEQ_0_NOT_FOUND =
+      "Async event queue \"%s\" not found";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__AEQ_0_DESTROYED =
+      "Async event queue \"%s\" destroyed";
+
+  @CliCommand(value = DESTROY_ASYNC_EVENT_QUEUE, help = DESTROY_ASYNC_EVENT_QUEUE__HELP)
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE)
+  public Result destroyAsyncEventQueue(
+      @CliOption(key = DESTROY_ASYNC_EVENT_QUEUE__ID, mandatory = true,
+          help = DESTROY_ASYNC_EVENT_QUEUE__ID__HELP) String aeqId,
+      @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
+          optionContext = ConverterHint.MEMBERGROUP,
+          help = DESTROY_ASYNC_EVENT_QUEUE__GROUP__HELP) String[] onGroups,
+      @CliOption(key = IFEXISTS, help = IFEXISTS_HELP, specifiedDefaultValue = "true",
+          unspecifiedDefaultValue = "false") boolean ifExists)
+      throws Throwable {
+    DestroyAsyncEventQueueFunctionArgs asyncEventQueueDestoryFunctionArgs =
+        new DestroyAsyncEventQueueFunctionArgs(aeqId, ifExists);
+
+    Set<DistributedMember> members = getMembers(onGroups, null);
```

Comment:
If you use getMembers() call, it will check for 0 instances found and throw an Exception which is handled by the executor to turn into a GemfireUserErrorResult, so you don't need to do the check at all.

Reply:
Good catch.